components/esp_modem/src/esp_modem_dce.cpp

```diff
@@ -88,6 +88,7 @@ bool DCE_Mode::set_unsafe(DTE *dte, ModuleIf *device, Netif &netif, modem_mode m
 {
     switch (m) {
     case modem_mode::UNDEF:
+    case modem_mode::DUAL_MODE: // Only DTE can be in Dual mode
```

Comment:
Question: When we construct the DTE in `DUAL_MODE`, which transitions are supported for `DCE` ? Could we get out of sync? How do we switch the second terminal to the `DATA_MODE`?

Reply:
The 2 terminals in USB modems are equal -> it does not matter which is used for data or commands.

So `set_mode(DATA)` connects the modem to cellular network: primary terminal is switched to data mode and swapped with secondary terminal.
And `set_mode(CMD)` disconnects the modem from cellular network.
CMUX is not supported (but not explicitly forbidden).

> Could we get out of sync?

In case ESP resets and connects to modem that is already in data mode, we will be out of sync. (I don't know how this can be solved in `esp_modem` :/)

Does it answer your questions?

Reply:
I see, thanks for explaining!

> (I don't know how this can be solved in esp_modem :/)

This is a general problem in `esp_modem` and we assume the initial mode is `UNDEF`, I've planned to implement the mode detection mechanism on startup, which would address this issue. But it's unrelated to your changes.